File: src/main/java/com/abysalto/middleware/source/dummyjson/dto/DummyProductList.java
package com.abysalto.middleware.source.dummyjson.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

/** DummyJSON paginated list envelope ({@code products}, {@code total}, {@code skip}, {@code limit}). */
@JsonIgnoreProperties(ignoreUnknown = true)
public record DummyProductList(
        List<DummyProduct> products,
        int total,
        int skip,
        int limit
) {
}

File: dotnet/src/Middleware.Infrastructure/Upstream/Dto/DummyProductList.cs
namespace Middleware.Infrastructure.Upstream.Dto;

/// <summary>DummyJSON paginated list envelope (<c>products</c>, <c>total</c>, <c>skip</c>, <c>limit</c>).</summary>
internal sealed record DummyProductList(
    IReadOnlyList<DummyProduct>? Products,
    int Total,
    int Skip,
    int Limit);
